pkg/streamingpromql/operators/aggregation.go

```diff
@@ -143,7 +143,6 @@ func (a *Aggregation) NextSeries(ctx context.Context) (types.InstantVectorSeries
 	// Iterate through inner series until the desired group is complete
 	for thisGroup.remainingSeriesCount > 0 {
```

Comment:
I think we could take this a step further, and move this entire `for` loop into its own method (I'm imagining something like `accumulateUntilGroupComplete`) - what do you think?